### PR TITLE
HotFix: Clear hive  database when logout 

### PR DESCRIPTION
**Pull request by nqhhdev**
### Issue

[ADR](https://github.com/linagora/twake-on-matrix/blob/a30abd7078afbc9bc40645993fee3b4ebcc2fd91/docs/adr/0006-dom-exception-error-when-delete-disk-for-web.md)

#### Web

https://github.com/linagora/twake-on-matrix/assets/99852347/d80eabe8-5cd0-471d-8c81-ae5aba02e4a3

https://github.com/linagora/twake-on-matrix/assets/99852347/48e9aab6-194e-43b0-8e67-a16b45f7396d

#### Mobile

https://github.com/linagora/twake-on-matrix/assets/99852347/cf3e42fb-8246-4e6a-b739-3792eb6a4135

https://github.com/linagora/twake-on-matrix/assets/99852347/1f5da0ff-03b8-4c5d-93ab-aa091d288878

### CODE REVIEW

File: lib/data/hive/hive_collection_tom_database.dart [hunk 1]
```diff
@@ -72,8 +73,11 @@ class HiveCollectionToMDatabase {
     try {
       await db.open();
     } catch (e, s) {
-      Logs().w('Unable to open Hive.', e, s);
+      Logs().w('Unable to open ToM Hive.', e, s);
+      const FlutterSecureStorage()
+          .delete(key: FlutterHiveCollectionsDatabase.cipherStorageKey);
       await db.clear().catchError((_) {});
+      await Hive.deleteFromDisk();
```
Comment (drminh2807) [inline lib/data/hive/hive_collection_tom_database.dart:80]: deleteFromDisk() then is it retry open to continue ?

File: lib/widgets/matrix.dart [hunk 2]
```diff
@@ -184,7 +179,7 @@ class MatrixState extends State<Matrix> with WidgetsBindingObserver {
         Logs().d('MatrixState::getLoginClient() Registering subs');
         _registerSubs(_loginClientCandidate!.clientName);
         _loginClientCandidate = null;
-        navigatorContext?.go('/rooms');
+        FluffyChatApp.router.go('/rooms');
```
Comment (drminh2807) [inline lib/widgets/matrix.dart:182]: can we change it to TwakeApp :)

File: lib/data/hive/hive_collection_tom_database.dart (hunk 1 above)
Comment (nqhhdev) [reply lib/data/hive/hive_collection_tom_database.dart:80]: back to old logic. we only handle this `clear()` func

File: lib/widgets/matrix.dart (hunk 2 above)
Comment (nqhhdev) [reply lib/widgets/matrix.dart:182]: Oke man, I'll update on last commit

Comment (nqhhdev) [reply lib/widgets/matrix.dart]: Done @drminh2807

File: lib/data/hive/hive_collection_tom_database.dart (hunk 1 above)
Comment (nqhhdev) [reply lib/data/hive/hive_collection_tom_database.dart:80]: Any question @drminh2807

**Review by drminh2807 — APPROVED** [review]

**Review by imGok — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**